core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/udf/UserDFOperator.java

```diff
@@ -0,0 +1,281 @@
+                Arrays.asList(PYTHON, pythonScriptPath, inputFilePath, outputFilePath));
+        
+        ProcessBuilder processBuilder = new ProcessBuilder(args);
+        processBuilder.redirectOutput(new File(pythonDebugOutputFilePath));
+        processPython = processBuilder.start();
+        return 1;
+    }
+    
+    public static void notifyPython(String pythonPID) throws IOException {
+        Runtime.getRuntime().exec("kill -SIGUSR2 " + pythonPID);
+    }
+    public void setInputOperator(IOperator operator) {
+        if (cursor != CLOSED) {
+            throw new TexeraException("Cannot link this operator to another operator after the operator is opened");
+        }
+        this.inputOperator = operator;
+    }
+    
+    @Override
+    public void close() throws TexeraException {
+        if (cursor == CLOSED) {
+            return;
+        }
+        if (inputOperator != null) {
+            inputOperator.close();
+        }
+        cursor = CLOSED;
+        return;
+    }
+    
+    @Override
+    public Schema getOutputSchema() {
+        return outputSchema;
+    }
+
+    @SuppressWarnings("restriction")
+    @Override
+    public void handle(Signal arg0) {
+        if (arg0.getNumber() == predicate.IPC_SIG) {
+            getPythonResult = true;
+        }
+    }
+    
+    @SuppressWarnings("restriction")
+    @Override
+    protected void setUp() throws TexeraException {
+        Schema inputSchema = inputOperator.getOutputSchema();
+        Signal.handle(new Signal(predicate.IPC_SIG_STRING), this);
+        handShake();
+        
+        outputSchema = inputSchema;
+    }
+
+    @Override
+    protected Tuple computeNextMatchingTuple() throws TexeraException {        
+        try {
+            Tuple inputTuple = inputOperator.getNextTuple();
+            //write attribute content to input mmap buffer
+            if (inputTuple == null) {
+                putTagIntoInputBuffer(predicate.TAG_NULL);
+            } else {
+                String inputTupleText = new ObjectMapper().writeValueAsString(inputTuple);
+                putTagIntoInputBuffer(String.valueOf((new ObjectMapper().writeValueAsString(inputTuple)).length()));
+                putJsonIntoInputBuffer(inputTupleText);
+            }
+
+            notifyPython( pythonPID.trim() );
+            
+            for( ; ; ) {
+                Thread.sleep(200);
+                if (getPythonResult) {
+                    getPythonResult = false;
+                    break;
+                }
+            }
+            //Output from buffer
+            String strLenTag = getTagFromOutputBuffer();
+            
+            if (strLenTag == predicate.TAG_WAIT) {
+                this.getNextTuple();
+            }
+            
+            if(strLenTag == predicate.TAG_NULL) {
+                processPython.destroy();
+                return null;
+            }
+            
+            String outputTupleJsonStr = getJsonFromOutputBuffer();
+            
```

Comment:
Why doesn't this function take output length as a parameter?

Reply:
Because the length of json text  is varied. And there will be other character in the text. To fully get the text, perhaps reading file content with a length of the jsom might be a good option.